LaunchAtLogin/LaunchAtLogin.swift

```diff
@@ -1,21 +1,156 @@
+import Combine
 import Foundation
 import ServiceManagement
 
-public struct LaunchAtLogin {
-	private static let id = "\(Bundle.main.bundleIdentifier!)-LaunchAtLoginHelper"
+public class LaunchAtLogin {
+    public static let observable = LaunchAtLogin()
+    public static let kvo = observable.kvo
 
-	public static var isEnabled: Bool {
-		get {
-			guard let jobs = (SMCopyAllJobDictionaries(kSMDomainUserLaunchd).takeRetainedValue() as? [[String: AnyObject]]) else {
-				return false
-			}
+    @available(macOS 10.15, *)
+    public static var publisher = observable.publisher
 
-			let job = jobs.first { $0["Label"] as! String == id }
+    public static var isEnabled: Bool {
+        get { observable.isEnabled }
+        set { observable.isEnabled = newValue }
+    }
 
-			return job?["OnDemand"] as? Bool ?? false
-		}
-		set {
-			SMLoginItemSetEnabled(id as CFString, newValue)
-		}
-	}
+    private let id = "\(Bundle.main.bundleIdentifier!)-LaunchAtLoginHelper"
+
+    public lazy var kvo = LaunchAtLoginKVO(launchAtLogin: self)
+
+    @available(macOS 10.15, *)
+    private lazy var _publisher = CurrentValueSubject<Bool, Never>(isEnabled)
+    @available(macOS 10.15, *)
+    public lazy var publisher = _publisher.eraseToAnyPublisher()
+
+    public var isEnabled: Bool {
+        get {
+            guard let jobs = (SMCopyAllJobDictionaries(kSMDomainUserLaunchd).takeRetainedValue() as? [[String: AnyObject]]) else {
+                return false
+            }
+
+            let job = jobs.first { $0["Label"] as! String == id }
+
+            return job?["OnDemand"] as? Bool ?? false
+        }
+        set {
+            if #available(macOS 10.15, *) {
+                objectWillChange.send()
+            }
+
+            kvo.willChangeValue(for: \.isEnabled)
+            SMLoginItemSetEnabled(id as CFString, newValue)
+            kvo.didChangeValue(for: \.isEnabled)
+
+            if #available(macOS 10.15, *) {
+                _publisher.send(newValue)
+            }
+        }
+    }
+}
+
+@available(macOS 10.15, *)
+extension LaunchAtLogin: ObservableObject { }
+
+// MARK: - LaunchAtLoginKVO
+extension LaunchAtLogin {
+    public class LaunchAtLoginKVO: NSObject {
```

Comment:
```suggestion
    public final class LaunchAtLoginKVO: NSObject {
```

Reply:
Are you sure it has to inherit from `NSObject` for it to be KVO-able?

Reply:
Yes, based on the [docs](https://developer.apple.com/documentation/swift/cocoa_design_patterns/using_key-value_observing_in_swift)
> You can only use key-value observing with classes that inherit from NSObject.